File: bot.js
from telegram import Bot, InlineKeyboardButton, InlineKeyboardMarkup, WebAppInfo
from telegram.ext import Updater, CommandHandler
import os

TOKEN = "8849881003:AAGk1D9_qWME23QXhfLn7q8Q-TTHo6RPxOY"
URL_MINI_APP = "https://riza875.github.io/Mini-app-guppy/"
URL_GAMBAR = "https://i.postimg.cc/cLvqV9zx/b668d18d-0522-4224-9a30-6e90701b02f0.png"

def start(update, context):
    keyboard = [[InlineKeyboardButton("🎮 Buka GUPPY.IO", web_app=WebAppInfo(url=URL_MINI_APP))]]
    
    update.message.reply_photo(
        photo=URL_GAMBAR,
        caption="🐟 **GUPPY.IO MINING** 🐟\n\nKlik tombol di bawah untuk mulai menambang!\n\n⭐ Dapatkan 50 poin per referral\n💰 1.000 poin = 1 USDT",
        parse_mode="Markdown",
        reply_markup=InlineKeyboardMarkup(keyboard)
    )

updater = Updater(TOKEN)
updater.dispatcher.add_handler(CommandHandler("start", start))
updater.start_polling()
print("Bot GUPPY.IO berjalan...")
updater.idle()

File: bot.py
import os
from telegram import Update, InlineKeyboardButton, InlineKeyboardMarkup, WebAppInfo, BotCommand
from telegram.ext import Application, CommandHandler, ContextTypes

# ========== HANDLER BOT ==========
async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user = update.effective_user

    keyboard = [[InlineKeyboardButton(
        "⛏️ Buka BTcloude Mining",
        web_app=WebAppInfo(url="https://mini-app-btcloude.vercel.app")
    )]]

    await update.message.reply_text(
        f"🪙 Halo {user.first_name}!\n\nKlik tombol di bawah untuk mulai menambang BTC:",
        reply_markup=InlineKeyboardMarkup(keyboard)
    )

async def help_cmd(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await update.message.reply_text(
        "⛏️ *BTcloude Mining Bot*\n\n"
        "/start — Buka aplikasi mining\n"
        "/help — Bantuan",
        parse_mode="Markdown"
    )

# ========== SETUP MENU COMMAND ==========
async def post_init(app: Application):
    await app.bot.set_my_commands([
        BotCommand("start", "🚀 Launch BTcloude App"),
        BotCommand("help", "❓ Bantuan"),
    ])
    print("Menu command telah diatur!")

# ========== MAIN ==========
def main():
    token = os.getenv("BOT_TOKEN")
    if not token:
        raise ValueError("BOT_TOKEN tidak ditemukan!")

    app = Application.builder().token(token).post_init(post_init).build()
    app.add_handler(CommandHandler("start", start))
    app.add_handler(CommandHandler("help", help_cmd))

    print("✅ BTcloude Bot berjalan...")
    app.run_polling()

if __name__ == "__main__":
    main()
